### PR TITLE
📖 Fix cert-manager version in the developer's guide

### PR DESCRIPTION
**Pull request by fmuyassarov**
**What this PR does / why we need it**:
Fix cert-manager version in the developer's guide

**Which issue(s) this PR fixes**
Fixes https://github.com/kubernetes-sigs/cluster-api/issues/4110

### CODE REVIEW

**Comment by fmuyassarov** [comment]
Does this change need to be backported in release 0-3 branch ?

**Review by enyinna1234 — COMMENTED** [review]
Hi @fmuyassarov, did you test this by going through the steps in the guide?

**Comment by fmuyassarov** [comment]
> Hi @fmuyassarov, did you test this by going through the steps in the guide?

Hi @enyinna1234 ,
Sorry for late reply.

Yes I did a quick test by using v0.16.0. However, I used Tilt setup. By default, Tilt setup is using [v0.16.1](https://github.com/kubernetes-sigs/cluster-api/blob/0008b5ba109e839067b96f4c5b52eab9fb657eca/tilt_modules/cert_manager/Tiltfile#L29) version, but it worked fine with v0.16.0 too. 
So, basically, I applied cert-manager (v0.16.0) and then ran the `tilt up` with docker provider. And as you can see all the three pods related  to cert-manager are up and running in `cert-manager `namespace.

```
$ kubectl get pods -A
NAMESPACE             NAME                                         READY   STATUS    RESTARTS   AGE
capd-system           capd-controller-manager-54d97dbf9d-558c9     2/2     Running   0          11m
capi-system           capi-controller-manager-77b7889ddb-q58sl     2/2     Running   0          11m
capi-webhook-system   capi-controller-manager-9556c79d6-vm6hn      2/2     Running   0          11m
cert-manager          cert-manager-cainjector-fc6c787db-z2lj7      1/1     Running   0          12m
cert-manager          cert-manager-d994d94d7-kwvzc                 1/1     Running   0          12m
cert-manager          cert-manager-webhook-845d9df8bf-62996        1/1     Running   0          12m
kube-system           coredns-74ff55c5b-d999d                      1/1     Running   0          21m
kube-system           coredns-74ff55c5b-wm48v                      1/1     Running   0          21m
kube-system           etcd-kind-control-plane                      1/1     Running   0          21m
kube-system           kindnet-5m59z                                1/1     Running   0          21m
kube-system           kindnet-bpjx6                                1/1     Running   0          21m
kube-system           kindnet-tkrq5                                1/1     Running   0          21m
kube-system           kube-apiserver-kind-control-plane            1/1     Running   0          21m
kube-system           kube-controller-manager-kind-control-plane   1/1     Running   0          21m
kube-system           kube-proxy-bw76d                             1/1     Running   0          21m
kube-system           kube-proxy-j2khw                             1/1     Running   0          21m
kube-system           kube-proxy-js28t                             1/1     Running   0          21m
kube-system           kube-scheduler-kind-control-plane            1/1     Running   0          21m
local-path-storage    local-path-provisioner-78776bfc44-l7zn2      1/1     Running   0          21m

```

File: docs/book/src/developer/guide.md [hunk 1]
```diff
@@ -82,7 +82,7 @@ The generated binary can be found at ./hack/tools/bin/envsubst
 You'll need to deploy [cert-manager] components on your [management cluster][mcluster], using `kubectl`
 
 ```bash
-kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v0.11.0/cert-manager.yaml
+kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v0.16.0/cert-manager.yaml
```
Comment (fabriziopandini) [inline docs/book/src/developer/guide.md:85]:
```suggestion
kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v0.16.1/cert-manager.yaml
```

Comment (fmuyassarov) [reply docs/book/src/developer/guide.md]: Fixed, thanks.

**Pull request merged**